charts/nr-k8s-otel-collector/templates/deployment-configmap.yaml

```diff
@@ -0,0 +1,334 @@
+
+      metricstransform/kube_pod_status_phase:
+        transforms:
+          - include: 'kube_pod_container_status_waiting'
+            match_type: strict 
+            action: update
+            new_name: 'kube_pod_container_status_phase'
+            operations:
+            - action: add_label
+              new_label: container_phase
+              new_value: waiting 
+          - include: 'kube_pod_container_status_running'
+            match_type: strict
+            action: update
+            new_name: 'kube_pod_container_status_phase'
+            operations:
+            - action: add_label
+              new_label: container_phase
+              new_value: running 
+          - include: 'kube_pod_container_status_terminated'
+            match_type: strict
+            action: update
+            new_name: 'kube_pod_container_status_phase'
+            operations:
+            - action: add_label
+              new_label: container_phase
+              new_value: terminated 
+
+      filter/exclude_zero_value_kube_pod_container_status:
+        metrics:
+          datapoint:
+            - metric.name == "kube_pod_container_status" and value_double == 0.0
+
+      resource/metrics:
+        attributes:
+          - key: host.id
+            from_attribute: host.name
+            action: upsert
+          # TODO chris: Upsert only when cluster name not found (resource detection override: true)
+          - key: k8s.cluster.name
+            action: upsert
+            value: {{ include "newrelic.common.cluster" . }}
+          - key: newrelicOnly
+            action: upsert
+            value: 'true'
+          - key: service.name
+            action: delete
+          - key: service_name
+            action: delete
+
+      batch:
+      
+      resource/events:
+        attributes:
+          - key: "event.name"
+            action: upsert
+            value: "InfrastructureEvent"
+          - key: "event.domain"
+            action: upsert
+            value: "newrelic-otel-event"
+          - key: "category"
+            action: upsert
+            value: "kubernetes"
+          - key: k8s.cluster.name
+            action: upsert
+            value: {{ include "newrelic.common.cluster" . }}
+          - key: newrelicOnly
+            action: upsert
+            value: 'true'
+
+      k8sattributes:
+        auth_type: "serviceAccount"
+        passthrough: false
+        filter:
+          node_from_env_var: KUBE_NODE_NAME
+        extract:
+          metadata:
+            - k8s.pod.name
+            - k8s.pod.uid
+            - k8s.deployment.name
+            - k8s.namespace.name
+            - k8s.node.name
+            - k8s.pod.start_time
+        pod_association:
+          - sources: 
+            - from: resource_attribute
+              name: k8s.pod.uid
+
+    exporters:
+      otlp:
+        endpoint: "nr-k8s-otel-collector-gateway:4317"
+        tls:
+          insecure: true
```

Comment:
I am confused, isn't this already the gateway?

Reply:
are you confused by the naming of that endpoint being gateway when its on the deployment (gateway) config? or why this endpoint exists?

Reply:
I mean the following:
 - this is the config of the deployment.
 - this OTLP exporter is sending data to `nr-k8s-otel-collector-gateway:4317`
 - `nr-k8s-otel-collector-gateway` is a service backed up by this deployment 🤔

Reply:
yes, this would have then sent the data through to the logs pipeline to write to NR, so that any changes that had to be made to _all_ the telemetry being sent through the gateway would go through that single pipeline, but we since had to make some changes to the structure of the collector because of memory issues (as in, right now all the data goes directly from the deployment and daemonset to NR without running through the other collectors) so now that paradigm of all the collected data running through a single pipeline that adds transforms needed on all the data is confusing; we have updated it. thanks :D